src/renderer/redux/reducers/rewards.js

```diff
@@ -62,24 +62,21 @@ reducers[ACTIONS.CLAIM_REWARD_STARTED] = (state, action) => {
 
 reducers[ACTIONS.CLAIM_REWARD_SUCCESS] = (state, action) => {
   const { reward } = action.data;
+  let { unclaimedRewards } = state;
 
-  const unclaimedRewardsByType = Object.assign({}, state.unclaimedRewardsByType);
-  const existingReward = unclaimedRewardsByType[reward.reward_type];
+  const index = unclaimedRewards.findIndex(ur => ur.reward_type === reward.reward_type);
+  unclaimedRewards = unclaimedRewards.slice(0, index).concat(unclaimedRewards.slice(index + 1));
```

Comment:
It looks like you're removing the current element?
I would suggest using splice instead.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

Reply:
That's likely because it's modifying the original array value and React only shallow checks.
If you return `[...unclaimedRewards]` instead of `unclaimedRewards`, it'll force the update.

This is a good use case for immutable.js as well

Reply:
Thanks. I did the same with claimed rewards and now they're showing up at the bottom like they should.